tests/benchmark/generate_csv.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import csv
+import json
+import os
+import sys
+
+
+def generate_csv(filename, tag):
+    """Extract function name, time used and memory usage from the metadata and write to the output CSV file.
+
+    Args:
+        filename (str): Filename of the performance json file to read
+        tag (str): Tag of the docker container
+    """
+    with open(filename) as input:
+        data = json.load(input)
+    benchmarks = data["benchmarks"]
+    runtime_version = os.path.basename(filename).split(".json")[0]
+
+    # Write data to CSV file
+    with open("{}.csv".format(os.path.splitext(filename)[0]), "wb") as output:
+        csv_writer = csv.writer(output, delimiter=',')
+        for benchmark in benchmarks:
+            try:
+                # Get the function name
+                func_name = benchmark["metadata"]["name"]
+                # Get the time used for this function, convert to millisecond
+                time_used = benchmark["runs"][0]["values"][0] * 1000
+                # Get the memory usage, convert to MB
+                mem_usage = benchmark["metadata"]["mem_max_rss"] / float(1<<20)
+                line = [tag, runtime_version, func_name, time_used, mem_usage]
+                # Write to CSV file
+                csv_writer.writerow(line)
+            except KeyError:
+                # Skip the benchmark result if it does not contain the fields we want
+                pass
+    output.close()
```

Comment:
You don't need to explicitly call output.close() since output was created in a 'with' statement.